### DISCUSSION TITLE
Screen Dump

### DISCUSSION THREAD

**Discussion by jojogib** [RomanKubiak/ctrlr#638] [Q&A]
Anyone knows how to get a screendump?
The format is: f0 00 20 32 00 16 36 (screendata 16*128) f7

Here is the whole sting after the dump:

f0 00 20 32 00 16 36 00 00 00 0c 0e 0f 03 0b 0b 07 0f 03 0b 0b 0b 0f 03 0f 03 0f 0e 0c 02 01 01 01 0d 05 05 05 01 09 05 05 09 01 01 01 02 0c 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 0f 07 0b 0b 07 0f 03 07 07 03 0f 03 0b 0b 07 0f 0b 03 0b 0f 03 0b 0b 0b 0f 03 07 0f 03 0f 07 0b 0b 07 0f 03 0b 0b 0b 0f 0f 00 00 00 0f 0f 0f 08 0e 0c 0b 0f 08 0a 0a 0b 0f 0c 0b 0c 0f 0f 0f 00 00 00 00 07 05 05 04 00 03 04 02 05 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 0f 08 0e 0e 08 0f 08 0f 0f 08 0f 08 0a 0a 0d 0f 0b 08 0b 0f 08 0a 0a 0b 0f 08 0f 0e 08 0f 0c 0b 0b 0d 0f 08 0a 0a 0b 0f 0f 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 01 0f 01 01 01 01 01 01 0d 05 05 09 01 05 0d 05 01 0d 05 05 05 01 0d 05 05 05 01 01 01 01 01 01 01 0f 01 09 05 05 05 01 0d 05 05 09 01 0d 05 05 09 01 0d 05 05 05 01 09 05 05 09 01 0d 05 05 09 01 0f 01 01 01 01 01 01 01 01 01 0d 05 05 09 01 0d 05 05 09 01 0d 01 0d 01 01 01 01 01 01 01 01 01 0f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 07 04 04 03 00 04 07 04 00 07 01 01 00 00 07 01 01 00 00 00 00 00 00 00 00 0f 00 04 05 05 02 00 07 01 01 00 00 07 01 03 04 00 07 05 05 04 00 07 01 01 07 00 07 04 04 03 00 0f 00 00 00 00 00 00 00 00 00 07 04 04 03 00 07 01 03 04 00 00 07 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 0c 02 02 01 01 01 01 03 02 0e 0c 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 0c 02 02 01 01 01 01 03 02 0e 0c 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 0c 02 02 01 01 01 01 03 02 0e 0c 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 08 08 00 00 0f 0f 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 0f 00 03 03 00 00 00 00 00 00 00 0f 0f 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 06 06 00 0f 0f 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 03 04 04 08 08 08 09 0d 0c 07 03 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 03 04 04 08 08 08 08 0c 0c 07 03 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 03 04 04 08 08 08 08 0c 0c 07 03 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0a 0a 02 0e 02 0a 02 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0f 00 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0a 02 0e 0e 02 0a 0a 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0f 00 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 02 0a 0a 0e 0e 0e 02 0a 02 0e 0e 0e 0e 02 0e 02 0a 0a 06 0e 0f 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0b 0a 08 0f 08 0b 08 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 00 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0b 08 0b 0f 0a 0a 08 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 00 0f 0f 0e 0e 0e 0f 0f 0f 0f 0f 0a 0a 08 0f 0b 0f 08 0b 08 0f 0d 0a 0a 08 0f 08 0a 0a 0d 0f 0f 00 0d 05 05 09 01 0d 05 05 09 01 0d 05 05 05 01 0d 05 05 09 01 0d 01 01 01 0d 01 0d 01 01 01 0e 00 00 00 00 0c 04 04 08 00 0c 04 04 04 00 08 04 04 08 00 08 04 04 08 00 0c 00 0c 00 00 00 00 0e 00 00 00 00 00 00 08 04 04 04 00 04 0c 04 00 04 04 0c 00 0c 04 04 04 00 00 00 00 00 00 00 00 0e 00 00 00 0c 04 04 04 00 0c 00 00 0c 00 00 00 0c 00 00 00 0c 00 0c 00 0c 00 00 00 00 00 00 00 0e 00 07 01 01 00 00 07 01 03 04 00 07 05 05 04 00 07 04 04 03 00 07 04 04 00 00 07 00 00 00 00 0f 00 00 00 00 07 04 04 03 00 07 05 05 04 00 03 04 04 02 00 07 01 01 07 00 00 07 00 00 00 00 00 0f 00 00 00 00 00 00 04 05 05 02 00 04 07 04 00 06 05 04 00 07 05 05 04 00 00 00 00 00 00 00 00 0f 00 00 00 07 01 01 00 00 06 01 01 06 00 00 00 07 04 04 00 03 04 03 00 07 04 04 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 0c 02 02 01 01 01 01 03 02 0e 0c 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 0c 02 0a 09 01 01 01 03 02 0e 0c 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 0c 02 0a 09 01 01 01 03 02 0e 0c 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 0c 02 02 01 01 01 01 03 02 0e 0c 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 0f 00 00 08 08 00 00 00 00 00 00 0f 0f 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 0f 00 00 01 01 00 00 00 00 00 00 0f 0f 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 0f 00 00 01 01 00 00 00 00 00 00 0f 0f 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 06 06 00 0f 0f 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 03 04 05 09 08 08 08 0c 0c 07 03 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 03 04 04 08 08 08 08 0c 0c 07 03 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 03 04 04 08 08 08 08 0c 0c 07 03 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 03 04 04 08 08 08 08 0c 0c 07 03 00 00 00 00 00 00 00 00 00 0f 00 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 0e 02 0a 02 0e 0e 0e 06 06 06 06 0e 0e 06 06 0e 0e 0f 00 00 00 00 00 00 04 0c 00 00 0c 04 04 00 0c 00 0c 00 00 00 08 08 08 08 00 00 08 08 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 04 04 0c 00 0c 04 0c 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 04 04 0c 00 00 00 0c 04 0c 00 00 00 00 0c 00 0c 04 04 08 00 0f 00 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 0f 08 0b 08 0f 0f 0f 08 0f 08 0f 08 0f 0a 09 0f 0f 0f 00 00 00 00 00 00 04 07 04 00 07 05 07 00 01 01 07 00 00 00 07 00 07 00 07 00 05 06 00 00 00 0f 00 00 00 00 00 00 00 00 00 00 00 04 05 07 00 07 04 07 00 00 00 00 00 00 00 00 00 00 00 00 00 0f 00 00 00 01 01 01 00 00 00 00 00 00 00 07 00 04 00 07 04 07 00 02 05 05 07 00 07 05 05 02 00 0f f7

Anyone knows how to interpret this in LUA? Thank you

**Comment by Ted-juh** [comment]
Could you answer the following questions please?

Which synth? Is there a sysex/ midi manual?What is it that you want to do?

**Reply by jojogib** [reply]
Its an FX unit from behringer which has the feature to "duplicate" the screen of the unit. So this large sysexdump above represents the pixel of the 16 x 128 monochrom screen?
There is no futher information in the manual. Is says: f0 00 20 32 00 16 36 (screendata 16*128) f7. After sending dump request I receive the large dump above.

**Comment by jojogib** [comment]
![20240503_175117](https://github.com/RomanKubiak/ctrlr/assets/95570341/1e41cfd1-5fbd-4ad9-a93b-5f6309988fe2)

Thats the matching screen for the dump

**Comment by Ted-juh** [comment]
16*128 = 2048 pixels or characters in some ASCII or other format.

If you ask me, pixels is unlikely because that should mean just 0 and 1. On or off. If it's characters then it's the question if 16 x 128 stands for height and width. As in 16 lines of 128 symbols/ characters. And each symbol/ character is an unit of pixels. If you take a good look, you'll see a lot of the same Hex codes. 

Should be doable but it's going to take a long time to figure out all the symbols and its Hex code.

**Reply by jojogib** [reply]
Ok thank you for your help. Its not really usefull for the ctrlr but i thought it would be a nice addition.

**Reply by Ted-juh** [reply]
Oh, don't get me wrong, it is a nice addition. I know Dnaldoog has some nice LCD screen replicas in some panels. 

Perhaps a weird question but could you provide a few pictures of the same screen in a way that the pixels are more visible. In the picture you posted, only the upper half is sharp enough to count the pixels. 

That way I can define how much pixels one character/ symbol is. Once I know that, it should be "easy" (cough) to figure out which Hex code represents a certain character/ symbol.

**Reply by Ted-juh** [reply]
The blocks of pixels are: 1 pixel wide and 4 pixels in height. 

![image](https://github.com/RomanKubiak/ctrlr/assets/71118185/bb9d8ff9-213b-4ec7-9505-643da1124f35)

The missing ones are the ones I couldn't figure out. And for a very good reason. And that is because I think the hex string you provided doesn't match the picture of the screen you posted. Because all that's certain now is that 00 is all off and 0F is all on. 

As you can see, there is no obvious order of the pixel arrangement if I correspond the hex string you provided with the screen dump you provided.

**Reply by jojogib** [reply]
Hey thank you. Do you think when I translate the whole thing in binary i could see the picture with 0 and 1 in with the right formation?
